crates/orderbook/src/ipfs.rs

```diff
@@ -53,11 +70,41 @@ impl Ipfs {
 
     fn prepare_url(&self, cid: &str) -> Url {
         let mut url = shared::url::join(&self.base, &format!("ipfs/{cid}"));
-        if let Some(query) = &self.query {
-            url.set_query(Some(query.as_str()));
+        if let Some(jwt) = &self.auth_token {
+            url.set_query(Some(&format!("pinataGatewayToken={}", jwt)));
         }
         url
     }
+
+    pub async fn add(&self, data: PinByHashRequest) -> Result<PinResponse> {
+        let url = format!("{}pinning/pinByHash", self.base);
+        // Prevent upload attempt when `auth_token is None`.
+        let jwt = if let Some(jwt) = &self.auth_token {
+            jwt
+        } else {
+            return Err(anyhow!("Can't post without valid JSON web token"));
+        };
+        let resp = self
+            .client
+            .post(url)
+            .header("Authorization", format!("Bearer {}", jwt))
+            .header("Content-Type", "application/json")
+            .json(&data)
+            .send()
+            .await?;
+
+        if resp.status().is_success() {
+            let data = resp.text().await?;
+            let response: PinResponse = serde_json::from_str(&data)?;
+            tracing::debug!("AppData pinned to IPFS with CID: {}", response.ipfs_hash);
```

Comment:
nit: We usually try to avoid dynamic log strings to make searching more consistent.
```suggestion
            tracing::debug!(cid = response.ipfs_hash, "AppData pinned to IPFS");
```

Reply:
Thanks! I had been wondering about what that was all about!